src/pages/home/report/PureReportActionItem.tsx

```diff
@@ -0,0 +1,1113 @@
+                event,
+                selection,
+                popoverAnchorRef.current,
+                reportID,
+                action.reportActionID,
+                originalReportID,
+                draftMessage ?? '',
+                () => setIsContextMenuActive(true),
+                toggleContextMenuFromActiveReportAction,
+                isArchivedRoom,
+                isChronosReport,
+                false,
+                false,
+                disabledActions,
+                false,
+                setIsEmojiPickerActive as () => void,
+                undefined,
+                isThreadReportParentAction,
+            );
+        },
+        [
+            draftMessage,
+            action,
+            reportID,
+            toggleContextMenuFromActiveReportAction,
+            originalReportID,
+            shouldDisplayContextMenu,
+            disabledActions,
+            isArchivedRoom,
+            isChronosReport,
+            isThreadReportParentAction,
+        ],
+    );
+
+    // Handles manual scrolling to the bottom of the chat when the last message is an actionable whisper and it's resolved.
+    // This fixes an issue where InvertedFlatList fails to auto scroll down and results in an empty space at the bottom of the chat in IOS.
+    useEffect(() => {
+        if (index !== 0 || !isActionableWhisper) {
+            return;
+        }
+
+        if (prevActionResolution !== (hasResolutionInOriginalMessage ? originalMessage.resolution : null)) {
+            reportScrollManager.scrollToIndex(index);
+        }
+    }, [index, originalMessage, prevActionResolution, reportScrollManager, isActionableWhisper, hasResolutionInOriginalMessage]);
+
+    const toggleReaction = useCallback(
+        (emoji: Emoji, ignoreSkinToneOnCompare?: boolean) => {
+            Report.toggleEmojiReaction(reportID, action, emoji, emojiReactions, undefined, ignoreSkinToneOnCompare);
+        },
+        [reportID, action, emojiReactions],
+    );
+
+    const contextValue = useMemo(
+        () => ({
+            anchor: popoverAnchorRef.current,
+            report: {...report, reportID: report?.reportID ?? ''},
+            reportNameValuePairs,
+            action,
+            transactionThreadReport,
+            checkIfContextMenuActive: toggleContextMenuFromActiveReportAction,
+            isDisabled: false,
+        }),
+        [report, action, toggleContextMenuFromActiveReportAction, transactionThreadReport, reportNameValuePairs],
+    );
+
+    const attachmentContextValue = useMemo(() => ({reportID, type: CONST.ATTACHMENT_TYPE.REPORT}), [reportID]);
+
+    const mentionReportContextValue = useMemo(() => ({currentReportID: report?.reportID ?? '-1'}), [report?.reportID]);
+
+    const actionableItemButtons: ActionableItem[] = useMemo(() => {
+        if (ReportActionsUtils.isActionableAddPaymentCard(action) && !doesUserHavePaymentCardAdded() && shouldRenderAddPaymentCard()) {
```

Comment:
`doesUserHavePaymentCardAdded` is also directly using Onyx data

Reply:
done